src/lib/operators/join_hash.cpp

```diff
@@ -527,114 +523,18 @@ class JoinHash::JoinHashImpl : public AbstractReadOnlyOperatorImpl {
-    output_chunks.reserve(expected_output_chunk_count);
-
-    // For every partition, create a reference segment.
-    auto partition_id = size_t{0};
-    auto output_chunk_id = size_t{0};
-    while (partition_id < build_side_pos_lists.size()) {
-      // Moving the values into a shared pos list saves us some work in write_output_segments. We know that
-      // build_pos_lists and probe_side_pos_lists will not be used again.
-      auto build_side_pos_list = std::make_shared<RowIDPosList>(std::move(build_side_pos_lists[partition_id]));
-      auto probe_side_pos_list = std::make_shared<RowIDPosList>(std::move(probe_side_pos_lists[partition_id]));
-
-      if (build_side_pos_list->empty() && probe_side_pos_list->empty()) {
-        ++partition_id;
-        continue;
-      }
+    auto create_left_side_pos_lists_by_segment =
+        (_build_input_table->type() == TableType::References && _output_column_order != OutputColumnOrder::RightOnly);
+    auto create_right_side_pos_lists_by_segment = (_probe_input_table->type() == TableType::References);
 
-      // If the input is heavily pre-filtered or the join results in very few matches, we might end up with a high
-      // number of chunks that contain only few rows. If a PosList is smaller than MIN_SIZE, we merge it with the
-      // following PosList(s) until a size between MIN_SIZE and MAX_SIZE is reached. This involves a trade-off:
-      // A lower number of output chunks reduces the overhead, especially when multi-threading is used. However,
-      // merging chunks destroys a potential references_single_chunk property of the PosList that would have been
-      // emitted otherwise. Search for guarantee_single_chunk in join_hash_steps.hpp for details.
-      constexpr auto MIN_SIZE = 500;
-      constexpr auto MAX_SIZE = MIN_SIZE * 2;
-      build_side_pos_list->reserve(MAX_SIZE);
-      probe_side_pos_list->reserve(MAX_SIZE);
-
-      // Checking the probe side's PosLists is sufficient. The PosLists from the build side have either the same
-      // size or are empty (in case of semi/anti joins).
-      while (partition_id + 1 < probe_side_pos_lists.size() && probe_side_pos_list->size() < MIN_SIZE &&
-             probe_side_pos_list->size() + probe_side_pos_lists[partition_id + 1].size() < MAX_SIZE) {
-        // Copy entries from following PosList into the current working set (build_side_pos_list) and free the memory
-        // used for the merged PosList.
-        std::copy(build_side_pos_lists[partition_id + 1].begin(), build_side_pos_lists[partition_id + 1].end(),
-                  std::back_inserter(*build_side_pos_list));
-        build_side_pos_lists[partition_id + 1] = {};
-
-        std::copy(probe_side_pos_lists[partition_id + 1].begin(), probe_side_pos_lists[partition_id + 1].end(),
-                  std::back_inserter(*probe_side_pos_list));
-        probe_side_pos_lists[partition_id + 1] = {};
-
-        ++partition_id;
-      }
+    auto allow_partition_merge = true;
```

Comment:
`constexpr` and add a comment? Might be good to reference the initial PR for the result merging.

Reply:
I referenced #2202 as a PR. Is the reference okay?

Reply:
Yes.